WooCommerce/Classes/ViewRelated/Editor/AztecEditorViewController.swift

```diff
@@ -0,0 +1,863 @@
+            placeholderLabel.leftAnchor.constraint(equalTo: richTextView.leftAnchor, constant: insets.left + richTextView.textContainer.lineFragmentPadding),
+            placeholderLabel.rightAnchor.constraint(equalTo: richTextView.rightAnchor, constant: -insets.right - richTextView.textContainer.lineFragmentPadding),
+            placeholderLabel.topAnchor.constraint(equalTo: richTextView.topAnchor, constant: insets.top),
+            placeholderLabel.bottomAnchor.constraint(lessThanOrEqualTo: richTextView.bottomAnchor, constant: insets.bottom)
+            ])
+    }
+
+    func registerAttachmentImageProviders() {
+        let providers: [TextViewAttachmentImageProvider] = [
+            SpecialTagAttachmentRenderer(),
+            CommentAttachmentRenderer(font: StyleManager.subheadlineBoldFont),
+            HTMLAttachmentRenderer(font: StyleManager.subheadlineBoldFont),
+            GutenpackAttachmentRenderer()
+        ]
+
+        for provider in providers {
+            richTextView.registerAttachmentImageProvider(provider)
+        }
+    }
+
+    func createToolbar() -> Aztec.FormatBar {
+        let toolbar = Aztec.FormatBar()
+
+        toolbar.tintColor = StyleManager.wooCommerceBrandColor
+        toolbar.highlightedTintColor = StyleManager.wooCommerceBrandColor.withAlphaComponent(0.5)
+        toolbar.selectedTintColor = StyleManager.wooSecondary
+        toolbar.disabledTintColor = StyleManager.buttonDisabledColor
+        toolbar.dividerTintColor = StyleManager.cellSeparatorColor
+        toolbar.overflowToggleIcon = Gridicon.iconOfType(.ellipsis)
+
+        updateToolbar(toolbar)
+
+        toolbar.barItemHandler = { [weak self] item in
+            self?.handleAction(for: item)
+        }
+
+        return toolbar
+    }
+
+    func updateToolbar(_ toolbar: Aztec.FormatBar) {
+        toolbar.trailingItem = nil
+
+        toolbar.setDefaultItems(scrollableItemsForToolbar,
+                                overflowItems: overflowItemsForToolbar)
+    }
+
+    func makeToolbarButton(identifier: FormattingIdentifier) -> FormatBarItem {
+        return makeToolbarButton(identifier: identifier.rawValue, provider: identifier)
+    }
+
+    func makeToolbarButton(identifier: String, provider: FormatBarItemProvider) -> FormatBarItem {
+        let button = FormatBarItem(image: provider.iconImage, identifier: identifier)
+        button.accessibilityLabel = provider.accessibilityLabel
+        button.accessibilityIdentifier = provider.accessibilityIdentifier
+        return button
+    }
+}
+
+private extension AztecEditorViewController {
+    func setHTML(_ html: String) {
+        editorView.setHTML(html)
+    }
+
+    func getHTML() -> String {
+        return editorView.getHTML()
+    }
+
+    func refreshPlaceholderVisibility() {
+        placeholderLabel.isHidden = richTextView.isHidden || !richTextView.text.isEmpty
+    }
+}
+
+// MARK: FormatBar Actions
+//
+extension AztecEditorViewController {
```

Comment:
I would explore a very lightweight implementation of the strategy pattern here. A set of classes, each of them encapsulating a different action (i.e. a class encapsulating toggleBold(), another one encapsulating toggleItalic(). all of them implementing an interface with two methods:
```
protocol AztecCommand {
func supports(id: FormatiingIdentifier) -> Bool
func execute(textView: RichTextView)
}
```

Another class that has just one method: `func handleAction(for barItem: FormatBarItem, in richView: AztecRichiView)` that loops through an array of implementations of the AztecCommand protocol, picks the one that returns true in the supports() method and class execute into it passing it the rich view.

Which is basically what Martin Fowler calls the [Production Rule System](https://martinfowler.com/dslCatalog/productionRule.html) 😊

Reply:
The reasons for that would be: first, you can remove a lot of code that does not really belong in the VC, and second, you have a set of highly cohesive rules that you can unit test very easily (and also write a simple integration test)